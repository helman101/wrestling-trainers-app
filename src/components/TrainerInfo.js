import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const TrainerInfo = (props) => {
  const { trainer, handleClick } = props;
  const { name, trainerImg } = trainer;

  return (
    <Link to="/trainerProfile" onClick={handleClick}>
      <figure>
        <img src={trainerImg} alt={name} />
        <figcaption>{name}</figcaption>
      </figure>
    </Link>
  );
};

TrainerInfo.propTypes = {
  trainer: PropTypes.shape({
    name: PropTypes.string.isRequired,
    trainerImg: PropTypes.string.isRequired,
  }).isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default TrainerInfo;
