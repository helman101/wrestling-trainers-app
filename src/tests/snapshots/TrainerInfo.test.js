import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import TrainerInfo from '../../components/TrainerInfo';

describe('TrainerInfo', () => {
  let component;
  let trainer;
  let clickMock;

  beforeEach(() => {
    clickMock = jest.fn();
    trainer = {
      name: 'Juan',
      trainerImg: 'https://www.industrialempathy.com/img/remote/ZiClJf-1920w.jpg',
    };
    component = renderer.create(
      <MemoryRouter>
        <TrainerInfo trainer={trainer} handleClick={clickMock} />
      </MemoryRouter>,
    );
  });

  it('should render propertly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
