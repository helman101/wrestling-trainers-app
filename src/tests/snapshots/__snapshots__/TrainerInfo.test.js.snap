// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainerInfo should render propertly 1`] = `
<a
  className="trainerInfo dFlex justifyContentCenter"
  href="/Trainer_profile"
  onClick={[Function]}
>
  <figure>
    <img
      alt="Juan"
      className="trainerImg"
      src="https://www.industrialempathy.com/img/remote/ZiClJf-1920w.jpg"
    />
    <figcaption>
      Juan
    </figcaption>
  </figure>
</a>
`;
